src/components/views/dialogs/AddressPickerDialog.js

```diff
@@ -510,6 +510,10 @@ module.exports = React.createClass({
         const AddressSelector = sdk.getComponent("elements.AddressSelector");
         this.scrollElement = null;
 
+        // Use set to avoid O(n*m) operation
+        const selectedAddresses = new Set(this.state.userList.map(({address}) => address));
```

Comment:
I think we should rename `userList` and `selectedAddresses` to `selectedList` and likewise rename `queryList` to `suggestedList`.

@t3chguy, wdyt? Worth the faff?

Reply:
done